### PR TITLE
json_normalize: Make code more pythonic and avoid modification of meta if mutable

### PR DESCRIPTION
**Pull request by davidfischer-ch**
- [x] closes itself #18610 
- [x] tests added / passed
- [x] passes `git diff master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

- Update json_normalize to prevent modifying `meta` parameter.
- Make the code more Pythonic too.

### CODE REVIEW

File: pandas/tests/io/json/test_normalize.py [hunk 1]
```diff
@@ -173,6 +173,20 @@ def test_meta_name_conflict(self):
         for val in ['metafoo', 'metabar', 'foo', 'bar']:
             assert val in result
 
+    def test_meta_parameter_not_modified(self):
+        data = [{'foo': 'hello',
```
Comment (jreback) [inline pandas/tests/io/json/test_normalize.py]: can you add the issue number here

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew not for 0.21.1

**Comment by davidfischer-ch** [comment]
@jreback is it OK now? Thanks.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**